### ISSUE TITLE
Front matter image processing assumes image as a single URL

### ISSUE THREAD

**Issue by KajKandler** [blurry-dev/blurry#47]
When defining the main image of an article as an "Imageobject" it breaks, because the code assumes the value to be a URL only.

```
if image := front_matter.get("image"):
   image_path = filepath.parent / Path(image)
```

This code needs to check for the frontmatter to be a url or to be an Image object and process the object accordingly. (An ImageObject furthermore contains other attributes that may need to be adapted).

Not sure how the thumbnailUrl is constructed and if it depends on the same assumption.

**Comment by johnfraney** [comment]
Good catch. I checked [BlogPosting](https://schema.org/BlogPosting) and `thumbnailUrl` is always just a URL rather than an object, but it would be good to support both image objects and URLs for `image`

![image](https://github.com/blurry-dev/blurry/assets/1728528/f0bc0733-5b18-4bcb-9cee-e92c5118806b)
